src/jsonp.coffee

```diff
@@ -0,0 +1,37 @@
+define [
+  'promise'
+  'helpers/url_helper'
+], (Promise, UrlHelper) ->
+  uid = 0
+
+  class JSONP
+    CALLBACK_PARAM = 'callback'
+
+    @fetch  = (url, data = {}) ->
+      promise = new Promise()
+      callback_name = JSONP._callbackName()
+      window[callback_name] = (response) -> promise.resolve(response)
+
```

Comment:
Consider removing this line and the one below `data[CALLBACK_PARAM] = callback_name` to make it more prominent that the return value is a promise.